### PR TITLE
Upgrade azure-iot-sdk-c 3.0

### PR DESCRIPTION
**Pull request by kimoantiqe**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [NA] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [NA] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [] Ready to merge

---

###### Summary 
Upgrade azure-iot-sdk for mariner 3.0

###### Change Log  
- Updated spec file
- Downloaded and generated tar for package manually
- Changed to tar.gz to match standard of all other packages (and upstream)

###### Does this affect the toolchain?  
NO

### CODE REVIEW

**Comment by anphel31** [comment]
I ran a buddybuild for this - https://dev.azure.com/mariner-org/mariner/_build/results?buildId=486950&view=results

**Comment by kimoantiqe** [comment]
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=494685&view=results

**Comment by kimoantiqe** [comment]
License did not change: https://github.com/Azure/azure-iot-sdk-c/blob/6b7753837ee18d2198759aa625d99081e255248f/LICENSE#L4

File: SPECS/azure-iot-sdk-c/azure-iot-sdk-c.spec [hunk 1]
```diff
@@ -86,6 +90,9 @@ install -p -m 755 provisioning_client/tools/tpm_device_provision/tpm_device_prov
 /usr/cmake/*
 
 %changelog
+*   Thu Jan 04 2024 Karim Eldegwy <karimeldegwy@microsoft.com> - 2023.08.07-1
+-   Upgrate to  2023.08.07-1
```
Comment (anphel31) [inline SPECS/azure-iot-sdk-c/azure-iot-sdk-c.spec:94]: No need to specify the release in this comment - just the version. Also an extra space is present. Suggest:
`Upgrade to 2023.08.07`

**Review by anphel31 — APPROVED** [review]

**Pull request merged**